src/bio/bio_xstream.c

```diff
@@ -749,8 +749,14 @@ bio_release_bdev(void *arg)
 	if (d_bdev->bb_desc == NULL)
 		return;
 
-	spdk_bdev_close(d_bdev->bb_desc);
-	d_bdev->bb_desc = NULL;
+	/*
+	 * It could be called from faulty device teardown procedure, where
```

Comment:
This wont affect if the device has been removed via DMG command, right? So I remove the device via DMG and did Hotplug which worked. Not sure if there is a way to put it back to NORMAL so dmg can list as Normal. Any way to do that?

Reply:
I'm not sure if I fully follow your question.

The situation here is that, if this bio_release_bdev() is called because of faulty teardown (device is still plugged), we keep the spdk bdev open; if this bio_release_bdev() is called because of hot remove teardown, we close the spdk bdev so that SPDK will be able to deconstruct the spdk bdev and ctrlr further.

To put a device (was marked as FAULTY before) back to NORMAL is called "revive device", it can be done through DMG device replace command (specify same device ID for old and new devices).